### DISCUSSION TITLE
Default language path

### DISCUSSION THREAD

**Discussion by Swepool** [ivanhofer/typesafe-i18n#321] [Q&A]
Is it possible to have "/' instead of "/lang" for default language? Checking out the package for sveltekit.

**Comment by ivanhofer** [comment]
Yes this is possible but will require you to store the locale information in a cookie or in SvelteKits session object in order to share it between the client and server

**Reply by Swepool** [reply]
> Yes this is possible but will require you to store the locale information in a cookie or in SvelteKits session object in order to share it between the client and server

Ah nice! Do you know any examples I can have a look at? 🤔

**Reply by ivanhofer** [reply]
maybe start with this tutorial: https://rodneylab.com/sveltekit-session-cookies/#desktop-sveltekit-session-cookie-code
I just glanced over it and it looks pretty straightforward.

1. First step would probably be: detecting the users preferred locale if no cookie is present. This can be done via the [locale detection feature](https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/detectors) of `typesafe-i18n`
2. store the locale in a cookie and session obbject
3. create an endpoint to switch locale (= resetting the cookie to another value)

Hope this helps :)

**Comment by Swepool** [comment]
hmmm think this is above my pay grade 😅 would be so nice to get rid of the 302.

**Reply by ivanhofer** [reply]
Oh, sorry I somehow missed this answer. Do you still need help?
